src/vs/workbench/contrib/search/browser/searchModel.ts

```diff
@@ -2004,21 +2130,33 @@ export class SearchModel extends Disposable {
 	}
 
 	async notebookSearch(query: ITextQuery, token: CancellationToken, onProgress?: (result: ISearchProgressItem) => void): Promise<{ completeData: ISearchComplete; scannedFiles: ResourceSet }> {
```

Comment:
Ideally the notebook search could be done in parallel with the basic text search

Reply:
Yes, that'd be ideal. I think there can be a lot more parallel code than what's there now, but we also need to have a list of open notebooks that the basic text search shouldn't find results for. Either that, or it does the whole search in parallel and then we dispose of results we don't need. I'll have to see what performs better.